pkg/util/util.go

```diff
@@ -140,3 +140,13 @@ func StripOutKeysFromJSONByteArray(jsonData []byte, keys []string) ([]byte, erro
 	resultJSON, _ := json.Marshal(jsonDataStruct)
 	return resultJSON, nil
 }
+
+// Filter returns a new slice holding only the elements of ss that satisfy test.
+func Filter[T any](ss []T, test func(T) bool) (ret []T) {
```

Comment:
Should we use a lib? https://github.com/samber/lo?tab=readme-ov-file#filter

Reply:
Uhh nice. 👍🏼

Reply:
I wonder if we should if it is just this one case.

Reply:
Yes, completely fair point, I wondered the same as I commented actually. I'm OK with having this as a one-off here.